File: iamport/iamport.py
#!/usr/bin/env python

# Echo client program
import socket
import sys
import time
import re

def tryToConnect( argv ):
    ip=argv[0]
    port=int(argv[1])

    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM )

    try:
        sock.connect( (ip, port) )
    except:
        print ip, port

    sock.close()

if __name__ == '__main__':

    stime=time.time()

    file = open(sys.argv[1])

    jobs = []

    while True:
        line = file.readline()

        if not line:
            break

        opts = re.split('\s*', line)

        if len(opts) > 2:
            job = tryToConnect( opts )
            jobs.append( job )
        else:
            print opts[0]

    file.close()

    etime=time.time()

    print 'elapsed %f' % (etime - stime)

File: iamport/iamport.js
#!/usr/bin/env node

//Echo client program

var net = require('net');
var fs = require('fs');

var stime=new Date().getTime();
var done=0;

function tryToConnect( ip, port ){
    done++;

    var socket = net.createConnection(port, ip);

    socket.on('error', function(err){
        console.log( ip + ' ' + port );
    }).on('connect', function(connect) {
        socket.destroy();
    }).on('close', function(had_error){
        done--;

        if(done == 0 ){
            var etime=new Date().getTime();

            console.log( "elapsed(ms) " + (etime - stime) );
        }
    });
}

fs.readFile(process.argv[2], 'utf-8', function(err, data){
    var lines = data.split('\n');

    lines.forEach(function(line){
        var opts=line.match(/[0-9\.]+/g);

        if( opts && opts.length > 1 ) {
            tryToConnect(opts[0], opts[1]);
        } else {
            console.log(line);
        }
    });
});


